networkx/tests/test_convert.py

```diff
@@ -279,3 +279,23 @@ class Custom(nx.Graph):
         # this raise exception
         # h._node.update((n, dd.copy()) for n, dd in g.nodes.items())
         # assert isinstance(h._node[1], custom_dict)
+
+
+def test_to_networkx_graph_non_edgelist():
+    invalid_edgelist = [1, 2, 3]
+    with pytest.raises(nx.NetworkXError, match="Input is not a valid edge list"):
+        nx.to_networkx_graph(invalid_edgelist)
+
+
+def test_to_dict_of_dicts_with_edge_data():
+    G = nx.Graph([(0, 1)])
+    expected = {0: {1: {"weight": 1.0}}, 1: {0: {"weight": 1.0}}}
+    assert nx.to_dict_of_dicts(G, edge_data={"weight": 1}) == expected
```

Comment:
edge_data should not be a container! The edges will end up all sharing the same container.

I think a test using the value `1` is a better choice. Also, see [this comment](https://github.com/networkx/networkx/issues/4305#issuecomment-719970727)

Reply:
Thanks for the feedback in the linked comment. I've moved the tests related to `edge_data` to #4321 which deals specifically with `edge_data`.